packages/backend/src/remote/activitypub/renderer/undo.ts

```diff
@@ -3,9 +3,11 @@ import { ILocalUser, User } from '@/models/entities/user.js';
 
 export default (object: any, user: { id: User['id'] }) => {
 	if (object == null) return null;
+	const id = typeof object.id === 'string' && object.id.startsWith(config.url) ? `${object.id}/undo` : undefined;
 
 	return {
 		type: 'Undo',
+		...(id ? { id } : {}),
```

Comment:
If we add the type above we can use `getApId`:
```suggestion

	return {
		type: 'Undo',
		id: getApId(object) + '/undo',
```

It has to be imported:
```ts
import { IObject, getApId } from '@/remote/activitypub/type.js';
```

I'm not sure if its necessary to check for `config.url`? It does not make sense to render an `Undo` for remote content and I don't think we are doing that anywhere. Even `renderFollowRelay` uses an ID created from the instance's URL.

Reply:
今はないが、id は optional なため。
ローカルチェックもねんのため。